builtin/providers/aws/resource_aws_config_config_rule.go

```diff
@@ -0,0 +1,311 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigConfigRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigConfigRulePut,
+		Read:   resourceAwsConfigConfigRuleRead,
+		Update: resourceAwsConfigConfigRulePut,
+		Delete: resourceAwsConfigConfigRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
You're 100% right, and it's in fact easier to make it required with a max validation.
Sorry for the noise here! :)